Fix vagrant support for Fedora 23

marun opened the pull request:
~~Fedora 21 is end-of-life as of 2015-12-01.  Updating to Fedora 23 to
ensure that those using vagrant+libvirt will be using a supported
release.  This change also updates the vagrant provision scripts to
support versions of Fedora other than 21.~~

Vagrant fixes for fedora 23
- wget is not installed by default on fedora 23.  Use curl instead since it is always available on recent Fedora.
- The repo url for cockpit resulted in an http redirect message being saved as the repo file which broke deployment.  Update the url to reflect the url that was redirected to.
- The main Fedora 23 repo includes salt packages, and there is no salt repo for 23. The salt bootstrap still creates a repo file for a nonexistent repo, though, and this change removes it to avoid having dnf report an error on every update.

This should also resolve #19834 

I've filed an issue for the salt bootstrapping problem: https://github.com/saltstack/salt-bootstrap/issues/736

marun commented:
cc: @derekwaynecarr @dcbw @knobunc

derekwaynecarr commented:
Thanks for the PR , there is another PR in flight that moved to fedora 23
specifically for vbox.  I want that to land before verifying this change.
Hopefully it will merge tonight.

On Tuesday, January 19, 2016, Kubernetes Bot <notifications@github.com>
wrote:

> GCE e2e test build/test *passed* for commit c068cd8
> <https://github.com/kubernetes/kubernetes/commit/c068cd8d1043ee186e5872960f36e3fb9ec28093>
> .
>
>    - Build Log
>    <https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/c068cd8d1043ee186e5872960f36e3fb9ec28093/kubernetes-pull-build-test-e2e-gce/24196/build-log.txt>
>    - Test Artifacts
>    <https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/c068cd8d1043ee186e5872960f36e3fb9ec28093/kubernetes-pull-build-test-e2e-gce/24196/artifacts/>
>    - Internal Jenkins Results
>    <http://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce//24196>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19841#issuecomment-173047960>
> .
>

marun committed:
Vagrant fixes for fedora 23

 - wget is not installed by default on fedora 23.  Use curl instead
   since it is always available on recent Fedora.

 - The repo url for cockpit resulted in an http redirect message being
   saved as the repo file which broke deployment.  Update the url to
   url that was redirected to and ensure that future redirects will be
   handled correctly.

 - The main Fedora 23 repo includes salt packages, and there is no
   salt repo for 23.  The salt bootstrap still creates a repo file for
   a nonexistent repo, though, and this change removes it to avoid
   having dnf report an error on every update.

marun commented:
@derekwaynecarr Assuming the network fixup is required for the opscode 23 boxes, this PR should now include only those fixes that #15998 was missing.

The one additional issue #15998 missed was the change in the cockpit repo url.  The url in HEAD results in a redirect message (to the url included in this PR) being saved as the repo file which causes cockpit installation and deployment to fail.

cc: @jayunit100 I've tested this to work with the official libvirt box.

marun commented:
Tested to work with vbox 5.x on os x.

dcbw commented:
LGTM; though for the salt thing, is it that the packages are *available* in Fedora through 'dnf install', or that they are already installed on the specific cloud images you're using in the Vagrantfile?

marun commented:
@dcbw Sorry, I can see how the commit message could be confusing. I meant to say that the packages are included in the official Fedora 23 repo, not that they are installed by default.  The salt bootstrap will configure a bad repo, and then when it installs packages those packages will come from the official repo.

dcbw commented:
@marun it looks like the cockpit repo got moved too, but your curl command doesn't follow the redirect.  You'll want to update the link and add "-L" to the curl on the master:

https://copr.fedorainfracloud.org/coprs/g/cockpit/cockpit-preview/repo/fedora-23/msuchy-cockpit-preview-fedora-23.repo

dcbw commented:
@marun hah, I see you've updated it already, so nevermind my comment.
